lib/posthog/feature_flags.rb

```diff
@@ -49,7 +50,24 @@ def load_feature_flags(force_reload = false)
     end
 
     def get_feature_variants(distinct_id, groups={}, person_properties={}, group_properties={})
+      decide_data = get_decide(distinct_id, groups, person_properties, group_properties)
+      if !decide_data.key?(:featureFlags)
+        raise DecideAPIError.new(decide_data.to_json)
+      else
+        stringify_keys(decide_data[:featureFlags] || {})
+      end
+    end
+
+    def get_feature_payloads(distinct_id, groups = {}, person_properties = {}, group_properties = {}, only_evaluate_locally = false)
+      decide_data = get_decide(distinct_id, groups, person_properties, group_properties)
+      if !decide_data.key?(:featureFlagPayloads)
+        raise DecideAPIError.new(decide_data.to_json)
```

Comment:
as in, okay to raise here, but catch in call site

Reply:
double check what happens when we just log here, as that means this function returns nil, not sure if we have existing  tests for this case 👀  & if it works as expected

Reply:
it will become a problem indeed for `get_feature_variants`, where we'll end up returning `nil` for get_all_flags, which shouldn't be possible (it's always a dict, either empty, or with things)

Reply:
I'm returning an empty hash on missing key error, not 100% sure how to test this 👀

Reply:
Actually looks like we're expecting a `nil` return from errors so it should be fine

Reply:
cool